### ISSUE TITLE
handling mehrerer Blocklisten

### ISSUE THREAD

**Issue by MarkyMarkDE** [haumacher/phoneblock#13]
ich hätte ja eigentlich ganz gerne die komplette Blocklist, sprich die aktuell 2532 aktiven SPAM-Nummern.

Wie könnte man das Problem seitens der Fritzbox handhaben, die ja nur maximal 2000 Einträge pro Telefonbuch zulassen?

Idee: wäre es möglich Deinem Server Zugriff auf die eigene Fritzbox zu geben, damit Du weitere Blocklisten anlegen und in die Anrufsperre eintragen kannst?
Sowas wie:

Blocklist1
Blocklist2
Blocklist3 
[...]

Man könnte das ganze ja mit einem speziellen Nutzer und Passwort machen, der dann die nötigen Rechte zugewiesen bekommt.
Sollten als Beispiel keine 3 Blocklisten mehr erforderlich sein, müsste Dein Server leere Listen natürlich auch löschen und die Anrufsperre löschen können.
Das ganze funktioniert natürlich nur, wenn es so oder so ähnlich technisch umsetzbar ist - ich trage nur Ideen vor :wink:

Die andere Idee ist Dir sicher nicht neu, man könnte Nummern, die zu einer "Telefonanlage" gehören, gruppieren, aber ich meine irgendwo gelesen zu haben, dass Du das noch nicht zum Laufen bekommen hast - richtig?
Damit könnte man sich vielleicht auch einige Einzeleinträge einsparen.

**Comment by haumacher** [comment]
Wenn [PhoneBlock](https://phoneblock.haumacher.de/) mehrere Blocklisten anbieten würde, dann könnte der Nutzer natürlich auch ganz einfach manuell mehrere Telefonbücher mit mehreren URLs anlegen und alle diese entsprechend der [PhoneBlock-Installationsanleitung](https://phoneblock.haumacher.de/phoneblock/setup.jsp) einrichten. Wenn es weniger als (Anzahl Blocklisten) * (Einträge pro Liste) Spam-Nummer gibt, dann wäre eben ein Telefonbuch leer, das wäre ja nicht schlimm.

> wäre es möglich Deinem Server Zugriff auf die eigene Fritzbox zu geben

Über das standardisierte TR-64-Protokoll ist das nicht möglich. Zugriff auf die Web-Oberfläche kann man zwar theoretisch über das Internet freigeben, das will ich aber nicht. Das hätte für die Nutzer ein zu hohes Risiko, wenn es auf dem [PhoneBlock-Server](https://phoneblock.haumacher.de/) doch einen "Sicherheits-Zwischenfall" geben würde.

> man könnte Nummern, die zu einer "Telefonanlage" gehören, gruppieren

Ja, das wäre meine präferierte Lösung. Allerdings kann man meines Wissens nach einer Rufnummer nicht ansehen, ab  welcher Stelle der Rufnummer eine "Durchwahl" an einem Anlagenanschluss anfängt. Um mehrere Rufnummern in einen Telefonbucheintrag zu quetschen, kann man glaube ich, entweder nur Rufnummern mit gemeinsamen Präfixen zu einem Eintrag zusammenfassen, oder aber einfach immer willkürlich 10 Nummern in einen Eintrag bündeln (die effizientere Lösung). 

In diesem Fall sieht man aber in der Anrufliste nicht mehr (oder nur noch indirekt) die tatsächlich beblockte Nummer, sondern immer erst nur den Namen des Telefonbucheintrags, in dem die Nummer enthalten war. Manche Leute möchten die Blockliste auch nicht zum Blockieren verwenden, sondern nur für die Anzeige der SPAM-Warnung. Das wird mit diesem Vorgehen auch erschwert.

**Comment by MarkyMarkDE** [comment]
> Wenn [PhoneBlock](https://phoneblock.haumacher.de/) mehrere Blocklisten anbieten würde, dann könnte der Nutzer natürlich auch ganz einfach manuell mehrere Telefonbücher mit mehreren URLs anlegen und alle diese entsprechend der [PhoneBlock-Installationsanleitung](https://phoneblock.haumacher.de/phoneblock/setup.jsp) einrichten. Wenn es weniger als (Anzahl Blocklisten) * (Einträge pro Liste) Spam-Nummer gibt, dann wäre eben ein Telefonbuch leer, das wäre ja nicht schlimm.

stimmt, wäre eigentlich nicht schlimm, es müssten dann halt jedes Mal alle Blocklisten, die der User hat, mit Deiner Datenbank abgeglichen werden.
> Über das standardisierte TR-64-Protokoll ist das nicht möglich. Zugriff auf die Web-Oberfläche kann man zwar theoretisch über das Internet freigeben, das will ich aber nicht. Das hätte für die Nutzer ein zu hohes Risiko, wenn es auf dem [PhoneBlock-Server](https://phoneblock.haumacher.de/) doch einen "Sicherheits-Zwischenfall" geben würde.

Daher wäre ja die Idee, das mit einem speziellen User zb. "PhoneBlock" zu machen, der dann halt nur gewisse Dinge tun darf und sonst nichts.
> Ja, das wäre meine präferierte Lösung. Allerdings kann man meines Wissens nach einer Rufnummer nicht ansehen, ab welcher Stelle der Rufnummer eine "Durchwahl" an einem Anlagenanschluss anfängt. Um mehrere Rufnummern in einen Telefonbucheintrag zu quetschen, kann man glaube ich, entweder nur Rufnummern mit gemeinsamen Präfixen zu einem Eintrag zusammenfassen, oder aber einfach immer willkürlich 10 Nummern in einen Eintrag bündeln (die effizientere Lösung).

Wenn die Rufnummer schon mit * in die Blockliste eingetragen wurde, wüsste man doch, dass danach eine Durchwahl kommt.
Aktuell müsste man allerdings den ganzen Datenbestand anhand gleicher Zahlenfolgen abgleichen, es gibt schon jetzt Nummern, die von der Vorwahl bis zu einer gewissen Stelle in der Rufnummer identisch sind, aber dann eine Durchwahl dran hängt

> In diesem Fall sieht man aber in der Anrufliste nicht mehr (oder nur noch indirekt) die tatsächlich beblockte Nummer, sondern immer erst nur den Namen des Telefonbucheintrags, in dem die Nummer enthalten war. Manche Leute möchten die Blockliste auch nicht zum Blockieren verwenden, sondern nur für die Anzeige der SPAM-Warnung. Das wird mit diesem Vorgehen auch erschwert.

das stimmt allerdings

**Comment by AlienLifeForm1** [comment]
Das mit der Abrufmöglichkeit mehrerer Blocklisten würde ich super finden.

Zu dem *: Ich habe bei mir im normalen Telefonbuch mehrere Einträge mit Stern für Firmen. In der Anruferliste und dem Display der Telefone steht beim Anruf Firmenname*234, wobei 234 die restlichen Stellen vom Anrufer sind. Wenn ich die Durchwahl im Kopf habe weiß ich gleich, wer mich von der Firma anruft. Wenn ich die Durchwahl nicht kenne habe ich zumindest die Zuordnung zur Firma. Wer die Blockliste nur für Warnungen verwendet hätte doch trotzdem genug Warnung und könnte dann in der FritzBox in der Anruferliste im Nachgang nachsehen, welche genaue Nummer angerufen hat.
